# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/apz/src/AsyncPanZoomController.cpp
# Commit: 243729640a19
# Full Hash: 243729640a19d0eb0a90a66af6eff0d43be56eb9
# Author: Hiroyuki Ikezoe <hikezoe.birchill@mozilla.com>
# Date: 2021-04-27 09:55:09
# Regressor Bug: 1707519
# File Overlap Count: 1
# Description:
#   Bug 1707519 - Add a null check to avoid crashes when mAnimation instance is not an OverscrollAnimation. r=botond
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D113292
# ==============================================================================

diff -r 3f6178ca6303 -r 243729640a19 gfx/layers/apz/src/AsyncPanZoomController.cpp
--- a/gfx/layers/apz/src/AsyncPanZoomController.cpp	Tue Apr 27 03:14:22 2021 +0000
+++ b/gfx/layers/apz/src/AsyncPanZoomController.cpp	Tue Apr 27 03:14:22 2021 +0000
@@ -2675,23 +2675,26 @@
   auto [logicalPanDisplacement, physicalPanDisplacement] =
       GetDisplacementsForPanGesture(aEvent);
 
-  if (mState == OVERSCROLL_ANIMATION &&
+  MOZ_ASSERT_IF(mState == OVERSCROLL_ANIMATION, mAnimation);
+  if (mState == OVERSCROLL_ANIMATION && mAnimation &&
       aFingersOnTouchpad == FingersOnTouchpad::No) {
     // If there is an on-going overscroll animation, we tell the animation
     // whether the displacements should be handled by the animation or not.
-    OverscrollAnimation* overscrollAnimation =
-        mAnimation->AsOverscrollAnimation();
-    overscrollAnimation->HandlePanMomentum(logicalPanDisplacement);
-    // And then as a result of the above call, if the animation is currently
-    // affecting on the axis, drop the displacement value on the axis so that we
-    // stop further oversrolling on the axis.
-    if (overscrollAnimation->IsManagingXAxis()) {
-      logicalPanDisplacement.x = 0;
-      physicalPanDisplacement.x = 0;
-    }
-    if (overscrollAnimation->IsManagingYAxis()) {
-      logicalPanDisplacement.y = 0;
-      physicalPanDisplacement.y = 0;
+    MOZ_ASSERT(mAnimation->AsOverscrollAnimation());
+    if (OverscrollAnimation* overscrollAnimation =
+            mAnimation->AsOverscrollAnimation()) {
+      overscrollAnimation->HandlePanMomentum(logicalPanDisplacement);
+      // And then as a result of the above call, if the animation is currently
+      // affecting on the axis, drop the displacement value on the axis so that
+      // we stop further oversrolling on the axis.
+      if (overscrollAnimation->IsManagingXAxis()) {
+        logicalPanDisplacement.x = 0;
+        physicalPanDisplacement.x = 0;
+      }
+      if (overscrollAnimation->IsManagingYAxis()) {
+        logicalPanDisplacement.y = 0;
+        physicalPanDisplacement.y = 0;
+      }
     }
   }
 
